source/Geant/proxy/test/test_BasicCpuTransport.cpp

```diff
@@ -475,12 +482,18 @@ OneStep(Track *track)
     /// Apply/do user actions
     /// ....
 
+    /// try to get a final track report
+    if (! IsAlive(*track)) {
+      std::cout<<"Last step for track "<< *track <<"\n";
+    }
+
     UpdateSwapPath(*track);
 
     /// Apply post step updates.
     //    ++track->fPhysicsState.fPstep;
-    track->fPhysicsState.fPstep = proposedPhysLength;
-    track->fStep = proposedPhysLength;
+    //track->fPhysicsState.fPstep = proposedPhysLength;
```

Comment:
Where are those now done?

Reply:
@mrguilima replied "Short answer to question is: fPstep gets updated on other places, so that particular line of update was not needed."

It actually sounds/look that we need
```
track->fPhysicsState.fPstep = 0;
track->fStep = 0;
```
isn't it?

Reply:
Both are done inside LinearStep, by subtracting step (argument given) from both fStep and fPstep.

Reply:
Humm .. okay .. I did not check (but would wander that that this is not true) that there is an inherent guaranteed that this will end up with zero values at the end of the step.